src/libaktualizr/asn1/asn1_message.h

```diff
@@ -0,0 +1,106 @@
+#ifndef ASN1_MESSAGE_H_
+#define ASN1_MESSAGE_H_
+#include <boost/intrusive_ptr.hpp>
+
+#include "AKIpUptaneMes.h"
+#include "AKTlsConfig.h"
+
+class Asn1Message;
+
+template <typename T>
+class Asn1Sub {
+ public:
+  Asn1Sub(boost::intrusive_ptr<Asn1Message> root, T* me) : root_(std::move(root)), me_(me) {}
+
+  T& operator*() const {
+    assert(me_ != nullptr);
+    return *me_;
+  }
+
+  T* operator->() const {
+    assert(me_ != nullptr);
+    return me_;
+  }
+
+ private:
+  boost::intrusive_ptr<Asn1Message> root_;
+  T* me_;
+};
+
+/**
+ * Reference counted holder for the top-level ASN1 message structure.
+ */
+
+class Asn1Message {
```

Comment:
It's called `Asn1Message` but it only covers `AKIpUptaneMes`

Reply:
I did play with the idea of making Asn1Message generic using some template magic, but it got rather out of hand.  For the purposes of RPC, I think we can keep all of the messages for the communication under a single ASN.1 message.  I suspect the answer is to rename `AKIpUptaneMes` to something more generic.